besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -858,6 +865,7 @@ void setBannedNodeIds(final List<String> values) {
   private BesuController<?> besuController;
   private StandaloneCommand standaloneCommands;
   private BesuConfiguration pluginCommonConfiguration;
+  private NodeKey nodeKey;
```

Comment:
If possible (not sure about subcommands) - it'd be nice if this class didn't own the NodeKey - only create it and pass it into the BesuControllerBuilder - it makes for nicer lifetime management (i.e. we don't have a 1/2 constructed BesuCommand class, which sometimes has a NodeKey, but sometimes not).

Reply:
the main requirement was subcommand, but since subcommand can access besuController, its a good idea to get it from there.